core/src/server/mod.rs

```diff
@@ -42,3 +42,65 @@ pub use helpers::{BatchResponseBuilder, BoundedWriter, MethodResponse, MethodSin
 pub use host_filtering::*;
 pub use rpc_module::*;
 pub use subscription::*;
+
+use jsonrpsee_types::{ErrorObjectOwned, ResponsePayload};
+
+/// Something that can be converted into a JSON-RPC method call response.
+///
+/// If the value couldn't be serialized/encoded, jsonrpsee will sent out an error
+/// to the client `response could not be serialized`.
+pub trait IntoResponse {
+	/// Output.
+	type Output: serde::Serialize + ToOwned<Owned = Self::Output>;
```

Comment:
Aah I see; it's a shame about the `ToOwned` but I see why!

`ToOwned` is impl for all `T where T: Clone`, so I wonder whether there `type Output: serde::Serialize + Clone` would be possible also and maybe a touch cleaner looking?. Hmm but the current approach is more "correct" though. Will have to ponder this, but yeah maybe it's best as it is here :)

Reply:
I don't know so I was strict and went with `ToOwned` and I don't really care.

Seems like it's equal to `Clone` in practice anyway.